nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/TrashedStateLifeCycleToPropertyMigrator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.trash;
+
+import static org.nuxeo.ecm.core.query.sql.NXQL.ECM_UUID;
+import static org.nuxeo.ecm.core.trash.PropertyTrashService.SYSPROP_IS_TRASHED;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.migration.MigrationService.MigrationContext;
+import org.nuxeo.runtime.migration.MigrationService.Migrator;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+/**
+ * Migrator of trashed state from lifecycle to property.
+ *
+ * @since 10.1 ??????
```

Comment:
what's with the `?????` ?

Reply:
a remember to change this if the ticket wasn't able to do feature freeze, thx